### ISSUE TITLE
Debugging doesn't work

### ISSUE THREAD

**Issue by frynio** [Microsoft/vscode-go#1135]
Okay, so I have a simple package, just wanted to try out the debugger from `vscode-go`.

Here's what I did:

![image](https://user-images.githubusercontent.com/15190180/29026911-702b7340-7b7e-11e7-9cb8-bcdf035acdef.png)

Now I want to Step Over. I click Step Over and this happens:

![image](https://user-images.githubusercontent.com/15190180/29026918-7d97c9f2-7b7e-11e7-92dc-1424ddc7ddfb.png)

Then I can press Step Over a few times, then it stops, and here's what I see in the console:

```
API server listening at: 127.0.0.1:39768
XD
[1 2 3 4 5]

=== Projekt PEA ===

=== MENU GŁÓWNE ===
unexpected fault address 0xc042cefd60
fatal error: fault
Exception 0xc0000096 0x0 0x0 0x455380
PC=0x455380

strconv.(*extFloat).frexp10(0x43878f, 0x4b6dbe, 0x5)
	C:/Go/src/strconv/extfloat.go:347 +0x60 fp=0xc04202fbd0 sp=0xc04202fb98
rax     0x4c5410
rbx     0x0
rcx     0x0
rdi     0xc04201e000
rsi     0xc04202fa01
rbp     0xc04202fba8
rsp     0xc04202fb98
r8      0x8fda8
r9      0xc04202fa78
r10     0x0
r11     0x246
r12     0x1
r13     0x27
r14     0x260
r15     0x200
rip     0x455380
rflags  0x10206
cs      0x33
fs      0x53
gs      0x2b
Failed to next.
Failed to next.
```
Without debugging it works perfectly. This is my `launch.json` (don't think it's needed here, but to cover all the basis):
```
{
    "version": "0.2.0",
    "configurations": [
    {
        "name": "Launch Package",
        "type": "go",
        "request": "launch",
        "mode": "debug",
        "program": "${workspaceRoot}"
    },
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            "remotePath": "",
            "port": 2345,
            "host": "127.0.0.1",
            "program": "${fileDirname}",
            "env": {},
            "args": [],
            "showLog": true
        }
    ]
}
```

Am I doing something wrong, or do you think that it is the issue with the plugin, or `vscode` itself?

**Comment by ramya-rao-a** [comment]
@frynio Can you try using `dlv` from the command line, add breakpoints and check if you are able to step over/in/out as expected?

**Comment by Jgeyen** [comment]
I had the exact same issue. I went into dlv and was able to replicate the issue with the unexpected fault. I was on Go version 1.8 and delve version 0.12.1. I updated Go to 1.8.3 and delve to 1.0 RC. That resolved the issue for me. I was able to step over some lines, but not others. For instance, I took the code from this site: https://blog.alexellis.io/golang-json-api-client/ and was able to step over the creation of the url, but not the  instantiation of the spaceClient. Also, if I put the breakpoint after the creation of the spaceClient, it gets to that breakpoint just fine. It is when I do a 'step' that the unexpected fault occurs.
